visual_behavior/ophys/response_analysis/cell_metrics.py

```diff
@@ -363,70 +365,144 @@ def compute_trace_metrics(traces):
 
     trace_metrics['ophys_experiment_id'] = ophys_experiment_id
-    trace_metrics['ophys_session_id'] = ophys_session_id
 
     trace_metrics = trace_metrics.reset_index()
     # trace_metrics = trace_metrics.melt(id_vars=['cell_specimen_id', 'ophys_experiment_id', 'ophys_session_id'])
     trace_metrics['condition'] = 'full_trace'
     trace_metrics['session_subset'] = 'full_session'
-    trace_metrics['stimuli'] = 'None'
-    trace_metrics['events'] = use_events
+    trace_metrics['stimuli'] = 'full_session'
+    trace_metrics['use_events'] = use_events
+    trace_metrics['filter_events'] = filter_events
+
+    if save:
+        filepath = get_metrics_df_filepath(ophys_experiment_id, condition='full_trace',
+                                           stimuli='full_session', session_subset='full_session',
+                                           use_events=use_events, filter_events=filter_events)
+        if os.path.exists(filepath):
+            os.remove(filepath)
+            print('h5 file exists for', ophys_experiment_id, ' - overwriting')
+        trace_metrics.to_hdf(filepath, key='df')
+        print('trace metrics saved for', ophys_experiment_id)
 
     return trace_metrics
 
 
-def generate_metrics_table(ophys_experiment_id, ophys_experiment_table, use_events=False, filter_events=False,
-                           condition='changes', session_subset='full_session', stimuli='pref_image'):
+def generate_cell_metrics_table(ophys_experiment_id, use_events=False, filter_events=False,
+                                condition='changes', session_subset='full_session', stimuli='pref_image', save=False):
+    """
+    Creates cell metrics table based on stimulus locked activity
+    Metrics include selectivity indices, mean image response, fano factor, fraction significant trials, etc
+    NOTE: session_subset = 'engaged' or 'disengaged' currently uses reward rate as engagement metric
+    The engagement_state from the behavior model will need to be added to the extended_stimulus_presentations table
+    in loading.get_ophys_dataset() to be able to be used here.
+    :param ophys_experiment_id: unique identifier for experiment
+    :param condition: 'changes', 'omissions', 'images'
+    :param stimuli: 'all_images', 'pref_image'
+    :param session_subset: 'engaged', 'disengaged', or 'full_session'
+    :param use_events: Boolean
+    :param filter_events: Boolean
+    :return:
+    """
+
+    print('generating cell metrics for', ophys_experiment_id)
 
-    ophys_session_id = ophys_experiment_table.loc[int(ophys_experiment_id)].ophys_session_id
     dataset = loading.get_ophys_dataset(ophys_experiment_id)
 
     analysis = ResponseAnalysis(dataset, use_extended_stimulus_presentations=True, use_events=use_events, filter_events=filter_events)
     sdf = analysis.get_response_df(df_name='stimulus_response_df')
 
     if condition == 'changes':
-        df = sdf[sdf.change == True]
+        df = sdf[sdf.is_change == True]
```

Comment:
Instead of writing `sdf[sdf.is_change == True]` you should just write `sdf[sdf.is_change]`. The column `is_change` is already a boolean, so you end up just computed `True == True` many times.

Reply:
Also I'm pretty sure using `df = sdf.query('is_change')` will be faster